### DISCUSSION TITLE
Does anyone else find insert/append a bit of a pain?

### DISCUSSION THREAD

**Discussion by eshrh** [meow-edit/meow#210] [General]
I have always felt like meow's insert/append idea to be one of the worse abstractions from vi.

I think part of it is because I spent a lot of time using vi and evil, but when I have a selection and I'm moving fast I get the choice between a and i wrong surprisingly often. Part of this is because the cursor obviously ends up in a different place based on the direction of movement. To insert after a character, one of the most common actions, you have to `find <char> append` but to do that in reverse you have to `negative-prefix find <char> insert`. This really annoys me sometimes, maybe because I have too much muscle memory of the `f char i` pattern.

The hack I use right now is to define another insert command that always inserts on the side of the selection the cursor is on. The idea is therefore to reverse the selection and then insert to get the opposite behavior. My rationale is that I will almost always want to insert TO where I moved, or else i wouldn't have moved there! Having to think and change between using i and a based on which direction I'm moving costs me time.

```emacs-lisp
(defun meow-insert-at-cursor ()
  (interactive)
  (if meow--temp-normal
      (progn
        (message "Quit temporary normal mode")
        (meow--switch-state 'motion))
    (meow--cancel-selection)
    (meow--switch-state 'insert)))
```
When I first switched to meow, I really did make an effort to learn to use insert/append, but I guess it just didn't stick.

Does anyone else have this problem? Or is it just me being bad at using meow?

**Comment by DogLooksGood** [comment]
I guess there's one have the similar feeling. 
https://github.com/meow-edit/meow/discussions/87.

Personally I don't have this issue, since I'm the author. Probably it's better to provide more insert commands.

**Comment by mclearc** [comment]
I like the basic idea of the two commands but I find that in practice I often have problems. For example, I'll be at the end of a line and use append and the cursor will be on the next line rather than at the end of current line if I had instead used insert. I have a lot of muscle memory from vim (and evil) so it may be that I just need to get used to the meow way.

**Comment by jasonperry** [comment]
Just want to say thanks for providing this solution! I was going to make a thread about the same thing. It is less confusing for me also to always insert at the cursor. It would be great if this command could become part of Meow.
